tools/scylla-sstable.cc

```diff
@@ -2750,6 +2754,242 @@ void shard_of_operation(schema_ptr schema, reader_permit permit,
+        init_cfg.tmp_dir_path = vm["temp-dir"].as<sstring>();
+    }
+
+    const auto format = get_output_format_from_options(vm, output_format::text);
+
+    do_with_cql_env_noreentrant_in_thread([&] (cql_test_env& env) mutable -> future<> {
+        auto& db = env.local_db();
+
+        const auto keyspace_name = "scylla_sstable";
+        co_await env.execute_cql(seastar::format("CREATE KEYSPACE {} WITH replication = {{'class': 'LocalStrategy'}}", keyspace_name));
+        auto& keyspace = db.find_keyspace(keyspace_name);
+
+        // Clone and modify the schema:
+        // * Change keyspace name to scylla_sstable
+        // * Generate a new ID
+        // * Drop all properties
+        //
+        // This will help avoid conflicts when querying sstables of system-tables
+        // and allows cql_test_env to work with a simple config (no EAR setup).
+        auto builder = schema_builder(keyspace_name, sstable_schema->cf_name());
+        for (const auto& col_kind : {column_kind::partition_key, column_kind::clustering_key, column_kind::static_column, column_kind::regular_column}) {
+            for (const auto& col : sstable_schema->columns(col_kind)) {
+                builder.with_column(col.name(), col.type, col_kind, col.view_virtual());
+
+                // Register any user types, so they are known by the time we create the table.
+                if (col.type->is_user_type()) {
+                    keyspace.add_user_type(dynamic_pointer_cast<const user_type_impl>(col.type));
+                }
+            }
+        }
+        auto schema = builder.build();
+
+        const auto table_name = schema->cf_name();
+
+        replica::schema_describe_helper describe_helper{db.as_data_dictionary()};
+
+        const auto original_schema_description = sstable_schema->describe(describe_helper, cql3::describe_option::STMTS_AND_INTERNALS);
+        const auto schema_description = schema->describe(describe_helper, cql3::describe_option::STMTS_AND_INTERNALS);
+
+        sst_log.debug("\noriginal schema:\n{}\nreplacement schema:\n{}\n\nNote: original keyspace name of {} was replaced with {}, original id of {} was replaced with {} and all properties were dropped!\n",
+                original_schema_description.create_statement.value(),
+                schema_description.create_statement.value(),
+                sstable_schema->ks_name(),
+                keyspace_name,
+                sstable_schema->id(),
+                schema->id());
+
+        co_await env.execute_cql(schema_description.create_statement.value());
+
+        auto& table = db.find_column_family(keyspace_name, table_name);
+
+        // We don't want to register the sstables with the table object,
+        // to avoid any attempt to compact/split/merge/rewrite them.
+        // Also, they were created with a foreign stable-manager (not part of
+        // cql_test_env).
+        // Use the virtual reader facility to isolate the sstables from
+        // cql-test-env.
```

Comment:
Additional positive side effect: bypasses the cache. We don't want the cache to grow on long queries, since we likely share the machine with a ScyllaDB instance.